utils/CMakeLists.txt

```diff
@@ -1,7 +1,16 @@
 get_target_property_required(PYTEST env PYTEST)
 
+# TODO: figure out a better way instead of this hack. Maybe add_test
+# just sucks or maybe we should write a wrapper test driver.
+add_test(
```

Comment:
CTest never builds targets.  The solution is something like:
make prepare_for_tests
make test

Reply:
That sucks and confirms my thought from the comment.

I saw that if the test uses a target added by `add_executable`[see `add_test` COMMAND](https://cmake.org/cmake/help/v3.10/command/add_test.html) it will use the once built.

Reply:
True, but it doesn't link the file dependency, just replaces the executable with the true path.

Reply:
https://stackoverflow.com/questions/733475/cmake-ctest-make-test-doesnt-build-tests

Reply:
This makes be conclude that ctest is not a good solution and that we should just wrap tests in shell script and call as normal custom targets.

Links related to open feature request to fix this in CMake:
 * [old mantis bug](https://cmake.org/Bug/view.php?id=8774)
 * [current tracking issue](https://gitlab.kitware.com/cmake/cmake/issues/8774)